trailblazer/apps/slurm/models.py

```diff
@@ -58,19 +59,16 @@ class SqueueResult(BaseModel):
     """Model used to parse SLURM squeue output."""
 
     jobs: List[SqueueJob]
-    jobs_status_distribution: Optional[Dict[str, float]]
+    jobs_status_distribution: Optional[Dict[str, float]] = None
 
-    @validator("jobs_status_distribution", always=True)
-    def set_jobs_status_distribution(
-        cls, _: Optional[List[SqueueJob]], values: Dict[str, Any]
-    ) -> Optional[Dict[str, float]]:
+    @model_validator(mode="after")
+    def set_jobs_status_distribution(self):
         """Set the fraction for each status present in the squeue jobs.
         Example {PENDING:0.33, RUNNING:0.33, FAILED: 0.33}."""
-        jobs: List[SqueueJob] = values["jobs"]
-        if not jobs:
+        if not self.jobs:
             raise MissingSqueueOutput("No squeue output")
-        status_distribution: Dict[str, float] = {}
-        job_statuses: List[str] = [job.status for job in jobs]
-        for job_status in set(job_statuses):
-            status_distribution[job_status] = round(job_statuses.count(job_status) / len(jobs), 2)
-        return status_distribution
+        job_statuses: List[str] = [job.status for job in self.jobs]
```

Comment:
We used the values parameter in the old method, I don't see why this would not work but we could also use something like in this [post](https://stackoverflow.com/questions/76734333/pydantic-v2-field-validator-values-argument-equivalent.

The only advantage I can think of is that the `FieldValidationInfo.data` property only contains previously validated fields. So if the `jobs` field had not been validated, it would not be accessible. Not sure how valuable that is. I think I prefer this approach

Reply:
It will be initialized and validated before being used in the model_validator so it should be safe. 👍